server/modules/storage/s3/definition.yml

```diff
@@ -1,11 +1,32 @@
 key: s3
 title: Amazon S3
 description: Amazon S3 is a cloud computing web service offered by Amazon Web Services which provides object storage.
-author: requarks.io
+author: andrewsim
 logo: https://static.requarks.io/logo/aws-s3.svg
 website: https://aws.amazon.com/s3/
+isAvailable: true
+supportedModes:
+  - push
```

Comment:
Clarification or documentation needed for `push`, `pull` and `sync` mode. I have only added these here to get rid of errors on the front-end.

Reply:
**Bi-directional (sync)**
In bi-directional mode, content is first pulled from the storage target. Any newer content overwrites local content. New content since last sync is then pushed to the storage target, overwriting any content on target if present.

**Push to target**
Content is always pushed to the storage target, overwriting any existing content. This is safest choice for backup scenarios.

**Pull from target**
Content is always pulled from the storage target, overwriting any local content which already exists. This choice is usually reserved for single-use content import. Caution with this option as any local content will always be overwritten! (edited) 

Only push must be implemented. Pull and sync are optional. They are mostly for the git module which can easily provide these capabilities.

Reply:
thanks for the clarification, i will leave **pull** and **sync** out for now. do give me a shout if it is needed in the future and i'm happy to open another pull request.